packages/code-infra/src/utils/git.mjs

```diff
@@ -0,0 +1,62 @@
+import { $ } from 'execa';
+import gitUrlParse from 'git-url-parse';
+
+/**
+ * Get current repository info from git remote
+ * @param {string[]} [remotes=['origin']] - Remote name(s) to check (default: ['origin'])
+ * @returns {Promise<{owner: string, repo: string}>} Repository owner and name
+ */
+export async function getRepositoryInfo(remotes = ['upstream', 'origin']) {
+  /**
+   * @type {{owner: string, repo: string} | undefined}
+   */
+  let result;
+  /**
+   * @type {Record<string, any>}
+   */
+  const cause = {};
+
+  for (let i = 0; i < remotes.length; i += 1) {
+    const remote = remotes[i];
+    try {
+      // eslint-disable-next-line no-await-in-loop
+      const cliResult = await $`git remote get-url ${remote}`;
+      const url = cliResult.stdout.trim();
+
+      const parsed = gitUrlParse(url);
+      if (parsed.source !== 'github.com' && parsed.owner !== 'mui') {
+        cause[remote] = { message: `Unsupported git remote URL: ${url}.` };
+        continue;
+      }
+
+      result = {
+        owner: parsed.owner,
+        repo: parsed.name,
+      };
+      break; // break as soon as we have a valid result
+    } catch (/** @type {any} */ error) {
+      const execaError = /** @type {import('execa').ExecaError} */ (error);
+      if (
+        i < remotes.length - 1 &&
+        typeof execaError.stderr === 'string' &&
+        execaError.stderr.includes('No such remote')
+      ) {
+        continue; // Try next remote
```

Comment:
we should flag an error here as well.

But I'm wondering, could this function be made faster and simpler if we built it on top of `git remote -v` to fetch all remotes at once with their url?